packages/ccd-js-gen/src/lib.ts

```diff
@@ -0,0 +1,269 @@
+        });
+
+        const dryRunClassDecl = sourceFile.addClass({
+            docs: [
+                `Smart contract client for dry running messages to a contract instance of '${contract.contractName}' on chain.`,
+            ],
+            isExported: true,
+            name: contractDryRunClassId,
+        });
+
+        classDecl
+            .addConstructor({
+                docs: ['Contruct a client for a contract instance on chain'],
+                parameters: [
+                    {
+                        name: grpcClientId,
+                        type: 'SDK.ConcordiumGRPCClient',
+                        scope: tsm.Scope.Public,
+                    },
+                    {
+                        name: contractAddressId,
+                        type: 'SDK.ContractAddress',
+                        isReadonly: true,
+                        scope: tsm.Scope.Public,
+                    },
+                ],
+            })
+            .setBodyText(
+                `this.${genericContractId} = new SDK.Contract(${grpcClientId}, ${contractAddressId}, '${contract.contractName}');
+this.${dryRunId} = new ${contractDryRunClassId}(this.${genericContractId});`
+            );
+
+        dryRunClassDecl.addConstructor({
+            docs: ['Contruct a client for a contract instance on chain'],
+            parameters: [
+                {
+                    name: genericContractId,
+                    type: 'SDK.Contract',
+                    scope: tsm.Scope.Private,
+                },
+            ],
+        });
+
+        for (const entrypointName of contract.entrypointNames) {
+            const transactionMetadataId = 'transactionMetadata';
+            const parameterId = 'parameter';
+            const signerId = 'signer';
+            classDecl
+                .addMethod({
+                    docs: [
+                        `Send a message to the '${entrypointName}' entrypoint of the '${contract.contractName}' contract.`,
+                    ],
+                    scope: tsm.Scope.Public,
+                    name: toCamelCase(entrypointName),
+                    parameters: [
+                        {
+                            name: transactionMetadataId,
+                            type: 'SDK.ContractTransactionMetadata',
+                        },
+                        {
+                            name: parameterId,
+                            type: 'SDK.HexString',
+                        },
+                        {
+                            name: signerId,
+                            type: 'SDK.AccountSigner',
+                        },
+                    ],
+                    returnType: 'Promise<SDK.HexString>',
+                })
+                .setBodyText(
+                    `return this.${genericContractId}.createAndSendUpdateTransaction(
+    '${entrypointName}',
+    SDK.encodeHexString,
+    ${transactionMetadataId},
+    ${parameterId},
+    ${signerId}
+);`
```

Comment:
Both receive functions and view functions are used to create methods here, thus creating helpers functions for signing/sending transactions for all of them. It would make more sense, if view functions would do `invokeView`.

Reply:
Right, I agree. The issue is that we can't really distinguish view functions from receive functions.

Reply:
Damn... well that sucks